commands/provider/service_provider.go

```diff
@@ -198,39 +200,16 @@ func (p *ServiceProvider) ServiceStop(id string) error {
 
 // ServiceValidate validates a service configuration and Dockerfile.
 func (p *ServiceProvider) ServiceValidate(path string) (string, error) {
-	validation, err := importer.Validate(path)
-	if err != nil {
+	if _, err := service.ReadDefinition(path); err != nil {
 		return "", err
 	}
-
-	if !validation.ServiceFileExist {
-		return fmt.Sprintf("%s File 'mesg.yml' does not exist", pretty.FailSign), nil
-	}
-
-	if len(validation.ServiceFileWarnings) > 0 {
-		var msg = fmt.Sprintf("%s File 'mesg.yml' is not valid. See documentation: https://docs.mesg.com/guide/service/service-file.html\n", pretty.FailSign)
-		for _, warning := range validation.ServiceFileWarnings {
-			msg += fmt.Sprintf("\t* %s\n", warning)
-		}
-		return msg, nil
-	}
-
-	if !validation.DockerfileExist {
-		return fmt.Sprintf("%s Dockerfile does not exist", pretty.FailSign), nil
-	}
-	if !validation.IsValid() {
-		return fmt.Sprintf("%s Service is not valid", pretty.FailSign), nil
-	}
-
-	return fmt.Sprintf(`%s Dockerfile exists
-%s mesg.yml is valid
-%s Service is valid`, pretty.SuccessSign, pretty.SuccessSign, pretty.SuccessSign), nil
+	return fmt.Sprintf(`%s mesg.yml is valid`, pretty.SuccessSign), nil
```

Comment:
Validate function should also check the Dockerfile

Reply:
No it shoudn't, because this is diffrent layer. 

Dockerfile is strictly related to Docker container and the service shouldn't be aware how the given manager starts a container. If you have multiple options to deploy service then `service` package shouldn't be responsible for validation if the service could be started. In that case, ContainerManager should be responsible and maybe we need to add `Validate(contextPath) error)` method to `Manager` Interface, but not to service itself.